packages/eslint-plugin/tests/rules/space-infix-ops.test.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable @typescript-eslint/internal/plugin-test-formatting */
+import rule from '../../src/rules/space-infix-ops';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('space-infix-ops', rule, {
+  valid: [
+    {
+      code: `
+        enum Test {
+          KEY1 = 2,
+        }
+      `,
+    },
+    {
+      code: `
+        enum Test {
+          KEY1 = "value",
+        }
+      `,
+    },
+    {
+      code: `
+        enum Test {
+          KEY1,
+        }
+      `,
+    },
+  ],
+  invalid: [
+    {
+      code: `
+        enum Test {
+          A= 2,
+          B = 1,
+        }
+      `,
+      output: `
+        enum Test {
+          A = 2,
+          B = 1,
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 12,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        enum Test {
+          KEY1= "value1",
+          KEY2 = "value2",
+        }
+      `,
+      output: `
+        enum Test {
+          KEY1 = "value1",
+          KEY2 = "value2",
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 15,
+          line: 3,
+        },
+      ],
+    },
+    {
+      code: `
+        enum Test {
+          A =2,
+          B = 1,
+        }
+      `,
+      output: `
+        enum Test {
+          A = 2,
+          B = 1,
+        }
+      `,
+      errors: [
+        {
+          messageId: 'missingSpace',
+          column: 13,
+          line: 3,
+        },
+      ],
+    },
+  ],
```

Comment:
Does it also work with:?

```ts
type Foo = number|string;
type Foo = number&string;
```

Perhaps it would be worth adding a test for those constructs as well

Reply:
Didn't think about this usecase. Will add it!

Reply:
I'd be okay with landing this as is and adding support for union/intersection as a separate PR.